apps/api/src/hyp3_api/validation.py

```diff
@@ -236,11 +232,44 @@ def check_opera_rtc_s1_static_coverage(job: dict, _) -> None:
 
     granule = granules[0]
     if not _has_opera_rtc_s1_static_coverage(granule):
-        raise GranuleValidationError(
-            f'Granule {granule} is outside the valid processing extent for OPERA RTC-S1 products.'
+        raise ValidationError(f'Granule {granule} is outside the valid processing extent for OPERA RTC-S1 products.')
+
+
+def check_dates_within_s1(job: dict, _) -> None:
+    date_params = {name: date.fromisoformat(value) for name, value in job['job_parameters'].items() if 'date' in name}
+
+    for param_name, date_value in date_params.items():
+        _validate_date_during_s1(param_name, date_value)
+
+
+def check_secondary_before_reference_date(job: dict, _) -> None:
+    def format_date(key: str) -> date:
+        return date.fromisoformat(job['job_parameters'][key])
+
+    reference, secondary = format_date('reference_date'), format_date('secondary_date')
+
+    if secondary >= reference:
+        raise ValidationError('secondary date must be earlier than reference date.')
```

Comment:
Can we just combine these two validator functions along these lines?

```suggestion
def check_aria_s1_gunw_dates(job: dict, _) -> None:
    def format_date(key: str) -> date:
        return date.fromisoformat(job['job_parameters'][key])

    reference, secondary = format_date('reference_date'), format_date('secondary_date')
    _validate_date_during_s1('reference_date', reference)
    _validate_date_during_s1('secondary_date', secondary)

    if secondary >= reference:
        raise ValidationError('secondary date must be earlier than reference date.')
```

Reply:
I kinda like it for testing, and it makes these validators a bit more generic but i don't mind combining them.